### PR TITLE
Custom model tutorial+notebook

### PR DESCRIPTION
**Pull request by exaexa**
Kinda finishes #137

### CODE REVIEW

File: docs/src/advanced/2_custom_model.md [hunk 1]
```diff
@@ -1,2 +1,151 @@
 
-# Custom models
+# Working with custom models
+
+It often happens that the intuitive representation of your data does not really
```
Comment (stelmo) [inline docs/src/advanced/2_custom_model.md:4]: It could happen

File: docs/src/advanced/2_custom_model.md [hunk 2]
```diff
@@ -1,2 +1,151 @@
...
+that work over the abstract type [`MetabolicModel`](@ref). To use your data
+structure in a model, you just make it a subtype of [`MetabolicModel`](@ref)
+and overload the required accessors. The accessors are functions that extract
+some relevant information, such as [`stoichiometry`](@ref) and
+[`bounds`](@ref), returning a fixed simple data type that can be further used
+by COBREXA.  You may see a complete list of accessors
+[here](../functions#Base-Types).
+
+A good solution to the second concern is a slightly more involved, as writing
+generic data modifiers is notoriously hard. Wtill, there is support for easily
```
Comment (stelmo) [inline docs/src/advanced/2_custom_model.md:27]: hard. Still,

File: docs/src/advanced/2_custom_model.md [hunk 3]
```diff
@@ -1,2 +1,151 @@
...
+functions such as [`with_added_reactions`](@ref) and [`with_set_bound`](@ref).
+
+!!! tip "Notebook available"
+    A better example of using a custom model structure is available
+    [in a separate notebook](../notebooks/8_custom_model.md).
+
+## Writing the generic accessors
+
+Let's write a data structure that represents a very small model that contains N
+metabolites that are converted in circle by N reactions. This may be useful for
```
Comment (stelmo) [inline docs/src/advanced/2_custom_model.md:38]: that are converted through linear, coupled reactions, e.g. A -> B -> C -> A with N= 3.

**Review by stelmo — APPROVED** [review]
Nice! The other advanced tutorials are not permanently deleted right?

**Comment by exaexa** [comment]
> Nice! The other advanced tutorials are not permanently deleted right?

no, I just didn't want to keep the stubs in live docs.

File: docs/src/advanced/2_custom_model.md (hunk 2 above)
Comment (exaexa) [reply docs/src/advanced/2_custom_model.md:27]: how did thiw get here?
:thinking:

**Pull request merged**